tests/client/query.ts

```diff
@@ -69,18 +78,122 @@ describe("Test Client Queries", () => {
     const inactiveEvents = await Events.eventQuery(connection)
       .filterByActive(true)
       .fetch();
-    assert.deepEqual(
-      [event1Pk],
-      inactiveEvents.map((event) => event.publicKey),
+    assert.ok(
+      inactiveEvents
+        .map((e) => e.publicKey.toBase58())
+        .includes(event1Pk.toBase58()),
     );
 
     const activeEvents = await Events.eventQuery(connection)
       .filterByActive(false)
       .fetch();
-    assert.deepEqual(
-      [event2Pk],
-      activeEvents.map((event) => event.publicKey),
+    assert.ok(
+      activeEvents
+        .map((e) => e.publicKey.toBase58())
+        .includes(event2Pk.toBase58()),
+    );
+  });
+
+  it("Fetch participants using query", async () => {
+    const program = anchor.workspace.ProtocolEvent as Program<ProtocolEvent>;
+    const connection = program.provider.connection;
+
+    const individual1 = await createIndividualParticipant(
+      program,
+      footballCategoryPda(),
+      "P1",
+      "Participant 1",
+    );
+    const individual2 = await createIndividualParticipant(
+      program,
+      footballCategoryPda(),
+      "P2",
+      "Participant 2",
+    );
+    const team1 = await createTeamParticipant(
+      program,
+      footballCategoryPda(),
+      "P3",
+      "Participant 3",
+    );
+
+    // fetch all participants
+
+    const allParticipants = await Participants.participantQuery(
+      connection,
+    ).fetch();
+
+    const unfilteredPks = allParticipants.map((participant) =>
+      participant.publicKey.toBase58(),
+    );
+    assert.ok(unfilteredPks.includes(individual1.toBase58()));
+    assert.ok(unfilteredPks.includes(individual2.toBase58()));
+    assert.ok(unfilteredPks.includes(team1.toBase58()));
+
+    // fetch participants by ParticipantType
+
+    const individualParticipants = await Participants.participantQuery(
+      connection,
+    )
+      .filterByParticipantType(new Individual())
```

Comment:
Interested in opinions on filtering by `ParticipantType`, this uses the new auto-gen client types which creates a new `class` for the individual enum values, rather than using an enum itself.

Classes defined here: https://github.com/BetDexLabs/protocol-event/blob/feat/client/client/types/ParticipantType.ts

`ParticipantTypeKind` (used in the query code) exported here; https://github.com/BetDexLabs/protocol-event/blob/feat/client/client/types/index.ts#L8

Reply:
One alternative I was thinking of is something like this:
```
  filterByParticipantType(participantType: string): Participants {
    switch (participantType) {
      case Individual.kind:
        this.filters.get("participantType").setValue(Individual.discriminator);
        break;
      case Team.kind:
        this.filters.get("participantType").setValue(Team.discriminator);
        break;
    }
    return this;
  }
```

So usage would be..
```
.filterByParticipantType("Individual")
```

Reply:
Feels like it just stops short of being useful. I knew there was something about the enum handling I didn't like. 

In the ideal case you'd be able to do something like 
```
filterByParticipantType(participantType: ParticipantTypeKind): Participants {
    this.filters.get("participantType").setValue(participantType.discriminator);
    return this;
}
```
but I don't think you can do something like that in Typescript. 

Maybe having something like I did for account querying will work here too?  👀

Reply:
This seems to pass type checking
```
interface EnumFilter {
  discriminator: number;
}

...

export class Participants extends AccountQuery<Participant> {
...
  filterByParticipantType(participantType: ParticipantTypeKind & EnumFilter): Participants {
    this.filters.get("participantType").setValue(participantType.discriminator);
    return this;
  }
...
}
```

Reply:
Still trying to figure out if we can pass in the Class type

Reply:
Stick with what you have/are happy with for now. I'll continue to ponder a nice way of accomplishing something _nicer_ here.